projects/igniteui-angular/src/lib/grids/grid-base.directive.ts

```diff
@@ -3835,26 +3835,8 @@ export abstract class IgxGridBaseDirective extends DisplayDensityBase implements
     }
 
     private createComponentInstance(component: any) {
-        // TODO: create component instance view viewContainerRef.createComponent(Component, Settings) overload
-        let dynamicFactory: ComponentFactory<any>;
-        const factoryResolver = this.moduleRef
-            ? this.moduleRef.componentFactoryResolver
-            : this.resolver;
-        try {
-            dynamicFactory = factoryResolver.resolveComponentFactory(component);
-        } catch (error) {
-            console.error(error);
-            return null;
-        }
-
-        const injector = this.moduleRef
-            ? this.moduleRef.injector
-            : this.injector;
-        const dynamicComponent: ComponentRef<any> = dynamicFactory.create(
-            injector
-        );
+        const dynamicComponent: ComponentRef<any> = createComponent(component, { environmentInjector: this.envInjector, elementInjector: this.injector } );
```

Comment:
@MayaKirova This, as far as I can tell, was the only place that was using the `moduleRef` (which was mostly wrong from the begging), so you should be able to remove the `private moduleRef: NgModuleRef<any>` injected prop entirely from all grids :)
And while at it, it also means this PR will technically also close https://github.com/IgniteUI/igniteui-angular/issues/12450 🥳